src/Application.cpp

```diff
@@ -0,0 +1,176 @@
+// Uncomment if opengl won't work
+/*
+#ifdef _WIN32
+extern "C" _declspec(dllexport) unsigned int NvOptimusEnablement = 0x00000001;
+#endif
+*/
+
+#include <chrono> // std::chrono::system_clock::now()
+
+#include "Application.h" // Implements
+#include "RenderView.h"
+
+// TEMP
+#include <fstream> // file reading
+void TEMP_setTexture(const char* filename, RenderView& rv);
+// END TEMP
+
+const std::string Application::DEFAULT_WINDOW_CAPTION = "Lame Boy";
+
+/*
+ * This function initializes sdl, creates a window and gets a OpenGL context. If an error occurs in this function
+ * the program should exit.
+ */
+void Application::initSDL() {
+    // Initialize SDL and check if SDL could be initialize.
+    assert(SDL_Init(SDL_INIT_VIDEO) >= 0);
+    atexit(SDL_Quit);
+    SDL_GL_LoadLibrary(nullptr); // Load the default OpenGL library
+
+    // Request an OpenGL 4.1 context and require hardware acceleration.
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 4);
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 1);
+    SDL_GL_SetAttribute(SDL_GL_ACCELERATED_VISUAL, 1);
+
+    // Try to put OpenGL in debug mode.
+    SDL_GL_SetAttribute(SDL_GL_CONTEXT_FLAGS, SDL_GL_CONTEXT_DEBUG_FLAG);
+
+    // Use double buffering. May be on by default. Not sure.
+    SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
+
+    // Create the window.
+    window = SDL_CreateWindow(DEFAULT_WINDOW_CAPTION.c_str(),
+                                          SDL_WINDOWPOS_UNDEFINED,
+                                          SDL_WINDOWPOS_UNDEFINED,
+                                          DEFAULT_WINDOW_WIDTH,
+                                          DEFAULT_WINDOW_HEIGHT,
+                                          SDL_WINDOW_OPENGL);
+    // Couldn't set video mode.
+    assert(window);
+
+    // Get gl context and set it to the current context for this window.
+    glContext = SDL_GL_CreateContext(window);
+    // Failed to create OpenGL context.
+    assert(glContext);
+
+    // Don't know if this is needed.
+    glewInit();
```

Comment:
GLEW manages extensions according to [this](https://www.opengl.org/sdk/libs/GLEW/) site
Do you use any extensions? If not then you could remove this line and remove GLEW

Reply:
`glewInit();` is required for making it work on my computer